extractor.go

```diff
@@ -339,6 +359,56 @@ func indexBlock(out Writer, sync *sync.Mutex, bh types.EventDataNewBlock) error
 	))
 }
 
+func indexValSetUpdates(out Writer, sync *sync.Mutex, updates *types.EventDataValidatorSetUpdates, height int64) error {
+	if len(updates.ValidatorUpdates) == 0 {
+		return nil
+	}
+
+	result := &codec.EventDataValidatorSetUpdates{}
+
+	for _, update := range updates.ValidatorUpdates {
+		nPK := &codec.PublicKey{}
+
+		switch update.PubKey.Type() {
+		case "ed25519":
+			nPK.Sum = &codec.PublicKey_Ed25519{Ed25519: update.PubKey.Bytes()}
+		case "secp256k1":
+			nPK.Sum = &codec.PublicKey_Secp256K1{Secp256K1: update.PubKey.Bytes()}
+		default:
+			return fmt.Errorf("unsupported pubkey type: %T", update.PubKey)
+		}
+
+		result.ValidatorUpdates = append(result.ValidatorUpdates, &codec.Validator{
+			Address:          update.Address.Bytes(),
+			VotingPower:      update.VotingPower,
+			ProposerPriority: update.ProposerPriority,
+			PubKey:           nPK,
+		})
+	}
+
+	data, err := proto.Marshal(result)
+	if err != nil {
+		return fmt.Errorf("cant marshal validator: %v", err)
+	}
+
+	sync.Lock()
+	defer sync.Unlock()
+
+	return out.WriteLine(fmt.Sprintf("%s %d %s",
+		dmValidator,
+		height,
+		base64.StdEncoding.EncodeToString(data),
+	))
+}
+
+// func formatFilename(name string) string {
```

Comment:
You can remove this part, it has been integrated into `Config`

Reply:
done